File: docs/versioned_docs/version-1.11.0/API-Reference/javascript-examples/api-monitor/update-session-id.js
const url = `${process.env.LANGFLOW_URL ?? ""}/api/v1/monitor/messages/session/01ce083d-748b-4b8d-97b6-33adbb6a528a?new_session_id=different_session_id`;

const options = {
  method: 'PATCH',
  headers: {
    "accept": `application/json`,
    "x-api-key": `${process.env.LANGFLOW_API_KEY ?? ""}`,
  },
};

fetch(url, options)
  .then(async (response) => {
    if (!response.ok) {
      throw new Error(`HTTP ${response.status}`);
    }
    const text = await response.text();
    console.log(text);
  })
  .catch((error) => console.error(error));

File: docs/versioned_docs/version-1.11.0/API-Reference/python-examples/api-monitor/update-session-id.py
import os

import requests

base = os.environ.get("LANGFLOW_URL", "")
api_key = os.environ.get("LANGFLOW_API_KEY", "")
flow_id = os.environ.get("FLOW_ID", "")

headers = {"accept": "application/json", "x-api-key": api_key}

list_resp = requests.get(
    f"{base}/api/v1/monitor/messages",
    headers=headers,
    params={"flow_id": flow_id},
    timeout=30,
)
list_resp.raise_for_status()
messages = list_resp.json()
if not messages:
    print("No messages; cannot migrate session id.")
    raise SystemExit(0)

old_session_id = messages[0]["session_id"]
new_session_id = f"{old_session_id}-migrated"

response = requests.patch(
    f"{base}/api/v1/monitor/messages/session/{old_session_id}",
    headers=headers,
    params={"new_session_id": new_session_id},
    timeout=30,
)
response.raise_for_status()
print(response.text)
